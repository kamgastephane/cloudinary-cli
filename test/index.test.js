require('..')
const file = require('../config/file')
let upload = require('../config/upload')
let update = require('../config/update')
let deleteFun = require('../config/delete')
let createTransform = require('../config/transform')

describe('env', () => {
  const data = {
    CLOUD_NAME: 'anything',
    API_KEY: 'anything',
    API_SECRET: 'anything'
  }
  it('get env file', () => {
    expect(file(data)).toMatchSnapshot()
  })
})

describe('upload', () => {
  it('return error if wrong path', () => {
    const path = ['./wrong-path']
    expect(upload(path)).toMatchSnapshot()
  })

  it('upload file successfully', () => {
    upload = jest.fn().mockImplementation(() => {
      console.log('uploaded')
      console.log(`publicID: bla
          width: 200
          height: 200
          url: https://bla.com`)
    })
    expect(upload('.files/path')).toMatchSnapshot()
  })
})

describe('update', () => {
  it('return error if wrong old public_id', () => {
    const oldPublicId = 'fake'
    expect(update(oldPublicId)).toMatchSnapshot()
  })

  it('successfully changes public_id', () => {
    update = jest.fn().mockImplementation((oldPublicId, newPublicName) => {
      console.log(`Public_id successfully changed from ${oldPublicId} to ${newPublicName}`)
    })
    expect(update('oldname', 'newNam')).toMatchSnapshot()
  })
})

describe('delete', () => {
  it('return error if wrong publicId', () => {
    const PublicId = ['fake']
    expect(deleteFun(PublicId)).toMatchSnapshot()
  })

  it('successfully delete file', () => {
    deleteFun = jest.fn().mockImplementation((publicId) => {
      console.log(`Successfully deleted ${publicId}`)
    })
    expect(deleteFun(['oldname'])).toMatchSnapshot()
  })
})

describe('createTransform', () => {
  it('successfully create a transformation', () => {
    createTransform = jest.fn().mockImplementation((name, parameters) => {
      console.log('transformation testTransform created!')
    })
    expect(createTransform('testTransform', 'w_100,h_200')).toMatchSnapshot()
  })
})
