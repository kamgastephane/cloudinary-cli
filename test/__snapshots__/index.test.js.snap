// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createTransform successfully create a transformation 1`] = `undefined`;

exports[`delete return error if wrong publicId 1`] = `undefined`;

exports[`delete successfully delete file 1`] = `undefined`;

exports[`env get env file 1`] = `undefined`;

exports[`update return error if wrong old public_id 1`] = `undefined`;

exports[`update successfully changes public_id 1`] = `undefined`;

exports[`upload return error if wrong path 1`] = `undefined`;

exports[`upload upload file successfully 1`] = `undefined`;
